src/utils/editor/source-search.js

```diff
@@ -119,6 +121,24 @@ function updateCursor(cm, state, keepSelection) {
   }
 }
 
+function getMatchIndex(state: Object, rev: boolean, nextMatch: Object) {
+  let matchIndex;
+  const count = state.results.length;
+
+  if (state.matchIndex === -1) {
+    return findIndex(state.results, nextMatch);
+  }
+
+  matchIndex = rev ? state.matchIndex - 1 : state.matchIndex + 1;
+  matchIndex = matchIndex % count;
+
+  if (matchIndex == 0) {
+    matchIndex = rev ? count : 1;
```

Comment:
Yeah have to agree here. In `SearchBar` we're displaying so the `+1` is necessary for UX but I don't think our index function should return out of range numbers. I'll look into it a bit further. 😄